packages/grid/_modules_/grid/models/api/componentsApi.ts

```diff
@@ -19,6 +19,14 @@ export interface ApiRefComponentsProperty extends GridIconSlotsComponent {
    * Header component rendered above the grid column header bar.
```

Comment:
How about this change too?

```suggestion
   * Prefer using the `Toolbar` slot. You should never need to use this slot. TODO remove.
```

It effectively creates a constraint to test if we can get away without it. While we don't have yet enough usage to know, I can guarantee you that with the type of scale we can expect from the component, if it's useful (solving a problem better than any alternative), developers will surface it. I think that we can reevaluate in 6 months.

Reply:
Agree, it will be easier to track.